### PR TITLE
Fix Uri modification and add unique telemetry id for preview

### PR DESCRIPTION
**Pull request by markcowl**
## Checklist

- [ ] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
This fix the issue https://github.com/Azure/azure-powershell/issues/9921

File: src/Accounts/Accounts/CommonModule/UserAgent.cs [hunk 1]
```diff
@@ -43,8 +53,13 @@ public class UniqueId
         /// <returns>Amended pipeline for retrieving a response</returns>
         public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken token, Action cancel, SignalDelegate signal, NextDelegate next)
         {
-            // add a header...
-            request.Headers.Add("x-ms-unique-id", Interlocked.Increment(ref this.count).ToString());
+            var userAgents = new ProductInfoHeaderValue[] { new ProductInfoHeaderValue("AzurePowerShell", $"Az4.0.0-preview") };
```
Comment (erich-wang) [inline src/Accounts/Accounts/CommonModule/UserAgent.cs]: @markcowl, do we expect to get "Az4.0.0-preview" userAgent in coming Az 2.x/3.x release?

**Comment by erich-wang** [comment]
@markcowl , do we need to modify ChangeLog.md here?

File: src/Accounts/Accounts/CommonModule/UserAgent.cs (hunk 1 above)
Comment (markcowl) [reply src/Accounts/Accounts/CommonModule/UserAgent.cs]: No, this is a separate pipeline, not used by existing cmdlets - only generated cmdlets use this pipeline

**Comment by markcowl** [comment]
@erich-wang yes, good catch, updated the changelog.  Pushed this change into a new PR https://github.com/Azure/azure-powershell/pull/10129

**Comment by markcowl** [comment]
@erich-wang closing in favor of #10129

**Pull request closed without merging**